server/master/src/main/java/org/apache/accumulo/master/tableOps/CreateDir.java

```diff
@@ -47,5 +58,21 @@ public void undo(long tid, Master master) throws Exception {
     VolumeManager fs = master.getFileSystem();
     fs.deleteRecursively(new Path(tableInfo.dir));
 
+    if (tableInfo.initialSplitSize > 0) {
+      SortedSet<Text> dirInfo = Utils
+          .getSortedSetFromFile(master.getInputStream(tableInfo.splitDirsFile), false);
+      for (Text dirname : dirInfo) {
+        fs.deleteRecursively(new Path(dirname.toString()));
+      }
+    }
+  }
+
+  private void createTabletDirectories(VolumeManager fs, SortedSet<Text> dirInfo)
+      throws IOException {
+
+    for (Text dir : dirInfo) {
+      if (!fs.mkdirs(new Path(dir.toString())))
```

Comment:
I wish HDFS had async or batch operations.  Could possibly use a thread pool here to speed up the case of many dirs. I think the HDFS client will batch request from multiple threads into a single RPC to namenode. This could be a follow on issue. Since multiple places in Accumulo code need to create or delete lots of files, could build these high level operations into volume manager.

Reply:
I created #631